### PR TITLE
Make React-Redux work in UMD/Global setup

### PR DESCRIPTION
**Pull request by kant2002**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

File: react-redux/react-redux.d.ts [hunk 1]
```diff
@@ -99,5 +103,9 @@ declare module "react-redux" {
   /**
    * Makes the Redux store available to the connect() calls in the component hierarchy below.
    */
-  export class Provider extends Component<ProviderProps, {}> { }
+  export class Provider extends __React.Component<ProviderProps, {}> { }
+}
+
+declare module "react-redux" {
+  export = ReactRedux;
```
Comment (seansfkelley) [inline react-redux/react-redux.d.ts]: Tests pass and this is the same pattern used by Redux and others, so AFAICT this should be a safe change.

**Comment by seansfkelley** [comment]
👍

**Comment by kant2002** [comment]
@vvakame could you merge this? it is approved by @seansfkelley

**Comment by tamird** [comment]
This change makes these typings incompatible with the typings shipped by Redux: https://github.com/reactjs/redux/blob/master/index.d.ts

**Comment by seansfkelley** [comment]
Redux ships a UMD build, so perhaps the right solution is to change the Redux typings themselves to express this as well?

More generally, I don't know what the methodology is for working with typings not defined in this repo. Presumably this will start to become irrelevant as typings are pushed to dedicated packages on npm (along with all the attendant changes), but in the interim it seems like it's going to be messy. Advice?

**Comment by tamird** [comment]
> Presumably this will start to become irrelevant as typings are pushed to dedicated packages on npm (along with all the attendant changes), but in the interim it seems like it's going to be messy.

I'm not sure this is going to happen. The ideal solution is for typings to travel with their packages; solutions like DT and npm-based distribution exist only to allow for typings external to the package.

In this case, if Redux's typings are incorrect, they definitely should be updated. I'm out of my depth on this, but perhaps @seansfkelley or @kant2002 could take that on?

**Pull request merged**